### ISSUE TITLE
Syntax error when there are some do-block in hash.

### ISSUE THREAD

**Issue by tuchida** [bbatsov/rubocop#4522]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

a.rb

```ruby
# frozen_string_literal: true

a = [1, 2, 3]

puts b: {
  c: a.map do |n|
    n * 2
  end,
  d: a.map do |n|
    n * 3
  end,
}
```

```sh
$ ruby a.rb 
{:b=>{:c=>[2, 4, 6], :d=>[3, 6, 9]}}
```

```sh
$ rubocop a.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Actual behavior

```sh
$ rubocop a.rb 
Inspecting 1 file
E

Offenses:

a.rb:9:12: E: unexpected token kDO_BLOCK
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  d: a.map do |n|
           ^^
a.rb:11:3: E: unexpected token kEND
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  end,
  ^^^

1 file inspected, 2 offenses detected
```

## Steps to reproduce the problem

When there are some do-block in hash, Rubocop throw syntax error.
If there is only one do-block, no error.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```sh
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
I can not reproduce this using `0.49.1`. Try upgrading to the latest version, and see if the problem persists.

**Comment by pocke** [comment]
We need the following `.rubocop.yml` to reproduce the problem.

```yaml
AllCops:
  TargetRubyVersion: 2.4
```

**Comment by pocke** [comment]
And I guess this problem is parser gem bug. https://github.com/whitequark/parser Could you report to the project?

**Comment by tuchida** [comment]
Thank you. I made that issue.
https://github.com/whitequark/parser/issues/353

**Comment by whitequark** [comment]
Root cause is https://github.com/whitequark/parser/issues/336.
